src/test/java/seedu/address/logic/commands/DeleteDeadlineCommandTest.java

```diff
@@ -0,0 +1,161 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static seedu.address.commons.core.Messages.MESSAGE_DELETE_DEADLINE_SUCCESS;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+import static seedu.address.testutil.TypicalProjects.getTypicalProjectsFolder;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.DateConversionException;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.project.Project;
+import seedu.address.model.task.deadline.Deadline;
+import seedu.address.testutil.DeadlineBuilder;
+import seedu.address.testutil.ProjectBuilder;
+
+public class DeleteDeadlineCommandTest {
+
+    private Model model;
+
+    @BeforeEach
+    public void setUp() throws DateConversionException {
+        model = new ModelManager(getTypicalAddressBook(), getTypicalProjectsFolder(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() throws DateConversionException {
+        Deadline deadlineToDelete = new DeadlineBuilder().build();
+        Project projectToEdit = model.getFilteredProjectList().get(INDEX_FIRST.getZeroBased());
+        Project editedProject = new ProjectBuilder(projectToEdit).build();
+        editedProject.addDeadline(deadlineToDelete);
+
+        model.setProject(
+                projectToEdit,
+                editedProject
+        );
+
+        Index lastDeadlineIndex = Index.fromOneBased(
+                model.getFilteredProjectList().get(INDEX_FIRST.getZeroBased()).getDeadlines().getDeadlines().size());
```

Comment:
Demeter's law! Probably find for now since we did similar things in other areas, but I think we should file an issue and 1 shot change all.